src/openrct2/world/Fountain.cpp

```diff
@@ -65,150 +65,140 @@ const uint8_t _fountainPatternFlags[] = {
     FOUNTAIN_FLAG::FAST                                                    // FAST_RANDOM_CHASERS
 };
 
-static int32_t jumping_fountain_get_type(const rct_jumping_fountain* jumpingFountain);
-static void jumping_fountain_continue(rct_jumping_fountain* jumpingFountain);
-static bool is_jumping_fountain(int32_t type, int32_t x, int32_t y, int32_t z);
-
-static void jumping_fountain_goto_edge(
-    const rct_jumping_fountain* jumpingFountain, int32_t x, int32_t y, int32_t z, int32_t availableDirections);
-static void jumping_fountain_bounce(
-    rct_jumping_fountain* jumpingFountain, int32_t x, int32_t y, int32_t z, int32_t availableDirections);
-static void jumping_fountain_split(
-    const rct_jumping_fountain* jumpingFountain, int32_t x, int32_t y, int32_t z, int32_t availableDirections);
-static void jumping_fountain_random(
-    const rct_jumping_fountain* jumpingFountain, int32_t x, int32_t y, int32_t z, int32_t availableDirections);
-static void jumping_fountain_create_next(
-    const rct_jumping_fountain* jumpingFountain, int32_t x, int32_t y, int32_t z, int32_t direction);
-
-void jumping_fountain_begin(int32_t type, int32_t x, int32_t y, const TileElement* tileElement)
+void JumpingFountain::Begin(const int32_t newType, const int32_t newX, const int32_t newY, const TileElement* tileElement)
```

Comment:
I would rather name this ```StartAnimation``` to clarify the purpose and ````Begin``` is somewhat unclear and usually used for iterators in C++